gradle/test.gradle

```diff
@@ -151,6 +151,8 @@ subprojects {
 
     forkEvery 1
     outputs.upToDateWhen{false}
+
+    dependsOn upgradeTest
```

Comment:
I think this should be moved to a CI change.  If upgradeTest is a separate source set it seems unexpected that the test tasks are dependent.

Reply:
I agree but for the short term rather than adding another CI task while we are revamping the CI this saves us some time. We can remove the depends after the CI refactor is done.